tests/core/engine_adapter/test_integration.py

```diff
@@ -853,6 +853,59 @@ def test_replace_query(ctx: TestContext):
         ctx.compare_with_current(table, replace_data)
 
 
+def test_replace_query_batched(ctx: TestContext):
```

Comment:
This was added to specifically test the issue users reported (and used to fail for some engines).